x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/settings/installation_button.tsx

```diff
@@ -95,19 +182,48 @@ export function InstallationButton(props: InstallationButtonProps) {
   );
 
   const updateButton = (
-    <EuiButton iconType={'refresh'} isLoading={isInstalling} onClick={handleClickUpdate}>
-      <FormattedMessage
-        id="xpack.fleet.integrations.updatePackage.updatePackageButtonLabel"
-        defaultMessage="Update to latest version"
-      />
-    </EuiButton>
+    <EuiFlexGroup alignItems="center">
+      <EuiFlexItem grow={false}>
+        <EuiButton
+          iconType={'refresh'}
+          isLoading={isInstalling}
+          onClick={upgradePackagePolicies ? toggleUpdateModal : () => handleClickUpdate(false)}
+        >
+          <FormattedMessage
+            id="xpack.fleet.integrations.updatePackage.updatePackageButtonLabel"
+            defaultMessage="Update to latest version"
+          />
+        </EuiButton>
+      </EuiFlexItem>
+      {packagePolicyCount > 0 && (
+        <EuiFlexItem grow={false}>
+          <EuiCheckbox
+            compressed
+            labelProps={{
+              style: {
+                display: 'flex',
+              },
+            }}
+            id="upgradePoliciesCheckbox"
+            checked={upgradePackagePolicies}
```

Comment:
does it make sense to have this checked by default?

Reply:
It's specced to be checked by default in the [parent issue](https://github.com/elastic/kibana/issues/106048)

> Add `Upgrade policies` checkbox next to upgrade package button if there is at least one policy for that package, it is checked by default

I think this _might_ be a bit too much potential risk to leave this checked by default. In cases where there are a lot of conflicts this might just add unexpected headaches for the user.

Reply:
we should push for an evergreen default experience - aka having this checked. after all, this is the only opportunity for the user to perform a bulk upgrade of their policies: if they proceed to upgrade the integration without this being checked, they can only perform the individual (manual) policy upgrade after that. I think the risk is mitigated by having the confirmation modal added here that shows a warning callout when there are conflicts.